# ReadFish command for flongle flow cell type
```bash
sudo /home/user/miniconda3/envs/readfish/bin/readfish targets --device MN37483 --experiment-name "RF Drosophila adaptive sampling" --toml Drosophila_chr_selection.toml --log-file dm_rf_2.log --channels 1 126 --paf-log pf_rf_chunk.log --chunk-log pf_rf_chunk.log
```
# ReadFish command for SpotOn flow cell type
```bash
sudo /home/user/miniconda3/envs/readfish/bin/readfish targets --device MN37483 --experiment-name "RF falciparum adaptive sampling" --toml falciparum_chr_selection_from_Homo.toml --log-file pf_rf.log --throttle 0.4 --paf-log pf_rf_chunk.log --chunk-log pf_rf_chunk.log 
```
# Guppy Basecall Server
```bash
/opt/ont/guppy/bin/guppy_basecall_server --log_path /var/log/guppy --config dna_r9.4.1_450bps_hac.cfg --num_callers 1 --cpu_threads_per_caller 62 --port /tmp/.guppy/5555 --ipc_threads 3 --device cuda:0:90%
```
# Location of Bulk fast5 file generated by local sequencing machine
```bash
/var/lib/minknow/data
```
# Selection of accepted read ids
```bash
grep -vFf unblocked_read_ids.txt sequencing_summary.txt | awk '{print $3}' > accepted_headers
```
# Select the accepted reads based on unblocked read ids
```bash
grep -wsf accepted_headers fastq_pass/combined.fastq -A 3 | sed '/^--$/d' > accepted_headers.fastq
```
## Make sure that the accepted_ids file does not contain @ symbol and pcregrep should be installed either via sudo apt or conda.

# Calculate per chromosome coverage
```bash
conda activate bbtools
pileup.sh in=target.bam out=coverage.txt reference=reference.fasta
```
```
sudo /home/user/miniconda3/envs/readfish/bin/readfish targets --device MN37483 --experiment-name "chr depletion adaptive sampling" --toml human_chr_depletion.toml --log-file hs_rf_deplete.log --throttle 0.4 --paf-log hs_rf_deplete_paf.log --chunk-log hf_rf_deplete_chunk.log --workers 16
```
# Simulating a minion run (through this you can playback two runs simultaneoulsy)
```bash
sudo service minknow stop
sudo /opt/ont/minknow/bin/mk_manager_svc --simulated-minion-devices N (N represents the number of minion devices you want to simulate)
Launch Minknow GUI and you will be seeing two devices.
For Bulk fast5 playback, edit the minknow sequencing toml file with simulation tag under custom_scripts. Change the device name and flow cell ID accordingly in minknow and readfish command.
```

